globus_automate_client/flows_client.py

```diff
@@ -349,7 +385,13 @@ def run_flow(
         :param flow_input: A Flow-specific dictionary specifying the input
             required for the Flow to run.
         """
-        ac = self._action_client_for_flow(flow_id, flow_scope)
+        if "authorizer" in kwargs:
```

Comment:
I think I like this too with the option of providing the kwarg. Not sure how this impacts my point above about a single authorizer set on the FlowsClient during __init__ so it doesn't have to get re-set elsewhere. Perhaps it ends up working?

Reply:
From what I understand so far, the flow list/update/deploy/delete operations require consent to the MANAGE_FLOWS scopes, and the flow-run,flow-action-status,flow-action-release, flow-action-cancel need consents to Flow-specific scopes. And to auto-discover the Flow-specific scopes, you need to have consented to the MANAGE_FLOWS scopes so that in the end we have two authorizers with two different sets of consents.

So it sort of seems like there's two authorizers at play. In this implementation, the init method expects the one with the MANAGE_FLOWS consents, and the authorizer_callback is used to pull the flow-dependant authorizers. Is that mental model right? Maybe there's something simpler we could be doing?

Reply:
I think you have it right. And, but this, all the operations on `MANAGE_FLOWS` is in the FlowsClient and we defer all the other operations to the ActionClient, so there's probably no need to juggle the authorizer within the FlowsClient (which is part of the goal of separating it this way).